bitsail-clients/bitsail-client-entry-flink/src/main/java/com/bytedance/bitsail/entry/flink/engine/FlinkEngineRunner.java

```diff
@@ -148,14 +172,49 @@ ProcessBuilder getRunProcBuilder(BitSailConfiguration jobConfiguration,
       flinkCommands.add(StringUtils.trim(property.getKey()) + "=" + StringUtils.trim(property.getValue()));
     }
 
-    flinkCommands.add(PackageResolver.getLibraryDir().resolve(ENTRY_JAR_NAME).toString());
-    flinkCommands.add("-xjob_conf");
-    flinkCommands.add(baseCommandArgs.getJobConf());
-
+    /*
+     * Customize path of BitSail JAR file for Kubernetes application mode, because Kubernetes
+     * application mode bundles BitSail JAR file together with the custom image and runs the user
+     * code's main() method on the cluster. Programmed path for
+     * BitSail JAR: local:///opt/flink/usrlibs/bitsail-core.jar
+     */
+    if (DEPLOYMENT_KUBERNETES_APPLICATION.equalsIgnoreCase(flinkRunCommandArgs.getDeploymentMode())) {
+      flinkCommands.add("local:///opt/flink/usrlibs/" + ENTRY_JAR_NAME);
```

Comment:
It's necessary to use `local` when specifying K8S cluster's jar files in Flink's command line, otherwise Flink won't find the path of jar. See reference in https://nightlies.apache.org/flink/flink-docs-release-1.11/ops/deployment/native_kubernetes.html#start-flink-application.